jwst/wfs_combine/wfs_combine_step.py

```diff
@@ -47,14 +46,18 @@ def process(self, input_table):
             # Do the processing
             output_model = wfs.do_all()
 
+            # The DataSet class does not close its resources.  Do that here.
+            wfs.input_1.close()
+            wfs.input_2.close()
+
             # Update necessary meta info in the output
             output_model.meta.cal_step.wfs_combine = 'COMPLETE'
             output_model.meta.asn.pool_name = asn_table['asn_pool']
-            output_model.meta.asn.table_name = op.basename(input_table)
+            output_model.meta.asn.table_name = os.path.basename(input_table)
 
             # Save the output file
             self.save_model(
                 output_model, suffix='wfscmb', output_file=outfile, format=False
             )
 
-        return None
+        return output_model
```

Comment:
So lines 59-61 are still in there to save the results to a file, as well as now passing `output_model` back to `stpipe`. Is `stpipe` going to try to save the model to a file too?

Reply:
That's a good question.  Let me check in the regression test run.

Reply:
Yes, it auto write out each result under the wrong name.  Lovely.  Let me fix this.

Reply:
So if I put it into a `if self.save_results:` block, it will then save it out properly. One then needs to supress the autosaving because there's the `return` statement.  Perhaps @stscieisenhamer has a better way of handling this.  I think this can be taken care of in issue #5511, as currently this step theoretically could return multiple products, but has no way to do so without utilizing a ModelContainer.

Reply:
What kind of wrong name is `stpipe` using when it saves to file? We should be able to tell it the proper product name suffix. @stscieisenhamer is the expert in that.

Reply:
The following:

```
$ ls /tmp/scratch/test_nircam_wfs_run_pipelines0/
config/
jw00632-o003_20191210t194815_wfs-image2_001_asn.json
jw00632-o003_20191210t194815_wfs-image3_001_asn.json
jw00632-o003_20191210t194815_wfs-image3_001_asn_wfscombine.fits <<<<<<<<<<<<
jw00632-o003_20191210t194815_wfs-image3_002_asn.json
jw00632-o003_20191210t194815_wfs-image3_002_asn_wfscombine.fits <<<<<<<<<<<<
jw00632-o003_t001_nircam_f212n-wlp8-nrca4_wfscmb-05.fits
jw00632-o003_t001_nircam_f212n-wlp8-nrca4_wfscmb-05_refine.fits
jw00632003002_03105_00001_nrca4_cal.fits
jw00632003002_03105_00001_nrca4_rate.fits
jw00632003002_03105_00002_nrca4_cal.fits
jw00632003002_03105_00002_nrca4_rate.fits
truth/
```

Reply:
The correct names are the `*wfscmb-05*.fits` ones which are generated from the name template in the association file.  How to get that into `meta.filename` of the returned model is not entirely clear to me.

Reply:
I believe the original code, where `return None` is used, was to handle the situation that the `stpipe` architecture did not handle multiple outputs. If there is not a reason to change this, I would suggest leaving it as-is.

However, if there is a need/desire to have the results returned, then the current way is as follows:

- Place each result in a `ModelContainer`
- For each resulting model, ensure that `result.filename = which_set['name']`
- ensure that the step parameter `self.output_use_mode = True`
- ensure that the step parameter `self.suffix = 'wfscb'`
- Remove the `save_model` statement

Ignoring possible other details being forgotten, this would fix the `return None`. IMHO, may want to leave the `return None` as-is.

Reply:
Though untested, theoretically the above setup should work when the return is a list of DataModels, without use a `ModelContainer`.  Did I say just leave the `return None` as-is?

Reply:
There's a very strong use case for returning something, as this is a step, and in order to test it, one needs the results of the step.

Reply:
And one also needs control over the `format` arg in `Step.save_model(format=False)`.  Without this, one gets:

```
jw00632-o003_t001_nircam_f212n-wlp8-nrca4_{suffix}-05_wfscmb.fits
```

Reply:
The formatting template in the association file is:

```
"name": "jw00632-o003_t001_nircam_f212n-wlp8-nrca4_{suffix}-05",
```

Reply:
So the current state of this PR produces the correct name, doesn't produce the incorrect name, and does so by hijacking the knowledge of when stpipe turns on `save_results`.  This is not ideal, but given that doing it properly would require a small refactor, I don't think we can do it before 7.7

Reply:
I think the only time this step would return multiple results is if you gave it an input ASN file that defines multiple products. This is not the case in normal operations (only 1 product per ASN).

Reply:
I added some comments to why it is implemented in this hacky way now.  In the future, perhaps we can refactor `Step.save_model` to expose `format` as a class or instance attribute.

Once tests pass I will merge.